src/Containers/FileUploadV2/useInformativePanels.ts

```diff
@@ -0,0 +1,209 @@
+    name: string;
+    /** worker; will do the job of reading the file */
+    worker: Worker | null;
+    /** the file associated with the informative panel */
+    file: File | null;
+}
+
+interface IInformativePanels {
+    /** array of panels */
+    panels: IPanel[];
+    /** names of files already uploaded, or failed, or cancelled */
+    makeItDisappear: string[];
+    /** names of files for which we want to start workers */
+    startWorkers: string[];
+}
+
+export const useInformativePanels = (
+    isTestIsFailure: boolean,
+    onFile: (base64StringFile: string) => void,
+    messageDuration: number,
+): readonly [
+    IPanel[],
+    (acceptedFiles: File[]) => void,
+    (name: string) => () => void,
+] => {
+    const isMounted = useMounted();
+    const [informativePanels, setInformativePanels] =
+        useState<IInformativePanels>({
+            panels: [],
+            makeItDisappear: [],
+            startWorkers: [],
+        });
+
+    /**
+     * set end state
+     */
+    const prepareForEndInformativePanel = useCallback(
+        (operationState: OperationState, informativePanel: IPanel): void => {
+            setInformativePanels((prev) => ({
+                ...prev,
+                panels: prev.panels.map((panel) => {
+                    const {name}=panel;
+                    if (name === informativePanel.name)
+                        return {
+                            ...panel,
+                            operationState,
+                        };
+                    return panel;
+                }),
+                makeItDisappear: [
+                    ...prev.makeItDisappear,
+                    informativePanel.name,
+                ],
+            }));
+        },
+        [],
+    );
+
+    /**
+     * terminate worker and set state of informative panel to success or failure and
+     * send order to remove informative panel in the future. also do whatever user
+     * wants to do with the file read in case of success
+     */
+    const onWorkerMessage = useCallback(
+        (e: any) => {
+            const { base64StringFile, name } = e.data;
+            if (base64StringFile === undefined) {
+                return;
+            }
+            const informativePanel = informativePanels.panels.find(
+                (panel) => panel.name === name,
+            );
+            if (informativePanel) {
+                if (
+                    base64StringFile === NO_BASE64STRINGFILE ||
+                    isTestIsFailure
+                ) {
+                    prepareForEndInformativePanel(
+                        OperationState.isFailure,
+                        informativePanel,
+                    );
+                } else {
+                    onFile(base64StringFile);
+                    prepareForEndInformativePanel(
+                        OperationState.isSuccess,
+                        informativePanel,
+                    );
```

Comment:
Duplicate set state just assign to variable and call setState at end of code to not repeat code